### ISSUE TITLE
Support protobuf 5.26.0

### ISSUE THREAD

**Issue by jonyscathe** [grpc/grpc#36142]
### Is your feature request related to a problem? Please describe.
Currently cannot update to protobuf 5.26.0 due to grpcio-tools dependencies.

### Describe the solution you'd like
grpcio-tools requirement for protobuf to be relaxed to protobuf>=4.21.3,<6.0dev
Not sure if any other changes are needed to support protobuf v26.0. The changes there don't look too drastic so I am guessing nothing (or not much).

### Describe alternatives you've considered
None - am currently sitting at protobuf v25.3 because of this

### Additional context
Poetry output from `poetry lock`:
```
Because grpcio-tools (1.62.1) depends on protobuf (>=4.21.6,<5.0dev)
 and test-project depends on protobuf (5.26.0), grpcio-tools is forbidden.
So, because test-project depends on grpcio-tools (1.62.1), version solving failed.
```

**Comment by sodul** [comment]
I suggest to remove the artificial upper bound limit on future versions of protobuf. As noted in this post:
https://discuss.python.org/t/how-to-address-formerly-unknown-incompatibilities/45873/5

> It’s not considered good practice for libraries to add upper bounds.

**Comment by CodingAnarchy** [comment]
This is especially confusing when I tried to update using `pip-tools`. Protobuf was updated to `5.26.1`, while `grpcio-tools` was *downgraded* to `1.30` to make it "work". But `1.30` does not work with the 3.11 version of Python!

**Comment by XuanWang-Amos** [comment]
Looks like the upper bound was changed from `<4.0dev` to `<5.0dev` in this PR two years ago as part of release process: https://github.com/grpc/grpc/pull/30377

**Comment by root-kidik** [comment]
+1

**Comment by Anton3** [comment]
On our macOS dev machines, core C++ `protobuf` package has been forcefully updated to 5.x via `brew`. So we need Python `protobuf` 5.x. But because of grpcio's incompatibility with 5.x, our builds are essentially broken on macOS, as of now. This issue blocks development for us.

**Comment by Anton3** [comment]
This whole situation has obviously been poorly handled by Protobuf + gRPC maintainers at Google. It should not have happened in the first place, and now this issue is at least 3 weeks old, without a fix

**Comment by jonyscathe** [comment]
Was fixed in another issue

**Comment by gnossen** [comment]
Hi all, thanks for chiming in on this issue. It's clear that this caused a lot of friction for a lot of people and we're dedicated to making it better. For starters, the first version that was originally planned to include Protobuf 5.X support was 1.64, 6 weeks later. We backported the update to 1.63 to get a solution out faster, which is now available as [1.63.0](https://pypi.org/project/grpcio-tools/1.63.0/).

But that's not enough. There have now been two major version bumps of the protobuf Python generated code API since gRPC hit 1.0 and both times the jump has caused gRPC users a lot of heartache. We've been working closely with the Protobuf team to institute a new policy where any new major version of the Protobuf Python package must _also_ support the generated code API of the _previous major version._ This will turn the hard installation errors or runtime errors that you're currently seeing with new major versions into deprecation warnings with a timeline of some number of months (TBD) to regenerate your `_pb2.py` files. We'll share more details once the new policy is finalized.
